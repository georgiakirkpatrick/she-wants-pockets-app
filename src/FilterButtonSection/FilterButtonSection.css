* {
    box-sizing: border-box;
}

.FilterButtonSection {
    width: 100%;
}

.FilterButtonSection div {
    width: 100%;
    display: flex;
    flex-direction: row;
    align-items: center;
}

.FilterButtonSection__button-section {
    justify-content: space-between;
}

.FilterButtonSection__button-section button {
    font-size: 18px;
    height: 50px;
    width: 48%;
    /* padding: .5em 3em .5em; */
    background-color: white;
    border-color: black;
    border-width: 2px;
}

.FilterButtonSection__tag-section {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    align-items: center;
    justify-content: flex-start;
    margin-block-start: 0;
    margin-block-end: 0;
    padding-inline-start: 0;
    margin: 1em 0 0;
}

.FilterButtonSection__tag-section li {
    font-size: 16px;
    padding: .5em 1em .5em;
    margin: 0 10px 10px 0;
    background-color: #e2e2e2;
    border-width: 0;
    border-radius: 50px;
}