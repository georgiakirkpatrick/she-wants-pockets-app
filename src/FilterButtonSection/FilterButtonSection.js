import React from 'react'
import './FilterButtonSection.css'

function FilterButtonSection(props) {
    const generateTags = props.selectedFilterOptions.map(option => (
        <li key={option.id} className='FilterButtonSection__tag'>
            {option.name}
        </li>
    ))
            
    return (
        <div className='FilterButtonSection'>
            <div className='FilterButtonSection__button-section'>
                <button>Filter</button>
                <button>Sort</button>
            </div>
            <ul className='FilterButtonSection__tag-section'>
                {generateTags}
            </ul>
        </div>
    )
}

FilterButtonSection.defaultProps = {
    selectedFilterOptions: []
}

export default FilterButtonSection