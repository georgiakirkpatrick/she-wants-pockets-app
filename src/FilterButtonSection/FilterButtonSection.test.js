import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import FilterButtonSection from './FilterButtonSection';

describe ('FilterButtonSection', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<FilterButtonSection />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it ('renders the UI as expected', () => {
        const tree = renderer
            .create(<FilterButtonSection />)
            .toJSON()
        expect(tree).toMatchSnapshot()    
    })
})