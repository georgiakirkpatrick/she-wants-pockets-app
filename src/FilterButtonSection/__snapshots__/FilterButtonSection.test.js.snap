// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterButtonSection renders the UI as expected 1`] = `
<div
  className="FilterButtonSection"
>
  <div
    className="FilterButtonSection__button-section"
  >
    <button>
      Filter
    </button>
    <button>
      Sort
    </button>
  </div>
  <ul
    className="FilterButtonSection__tag-section"
  />
</div>
`;
